src/select.rs

```diff
@@ -104,25 +144,56 @@ impl<T: Clone> PromptInteraction<T> for Select<T> {
     fn render(&mut self, state: &State<T>) -> String {
         let theme = THEME.lock().unwrap();
 
-        let line1 = theme.format_header(&state.into(), &self.prompt);
+        let header_display = theme.format_header(&state.into(), &self.prompt);
+        let items_display: String;
+        let footer_display = theme.format_footer(&state.into());
 
-        let line2: String = self
-            .items
-            .iter()
-            .enumerate()
-            .map(|(i, item)| {
-                theme.format_select_item(&state.into(), self.cursor == i, &item.label, &item.hint)
-            })
-            .collect();
-        let line3 = theme.format_footer(&state.into());
+        if self.switch_filter_mode {
+            let filter_regex = Regex::new(&format!("(?i){}", self.filter))
+                .unwrap_or_else(|_| Regex::new(r"^\b$").unwrap());
```

Comment:
I personally might like to switch between "prefix-match", "wildcard-match" and "fuzzy-match" for different scenarios (and list-length, as wildcard+fuzzy will be slower for very long lists). It wouldn't be wrong either to allow providing an own `FnOnce(&str) -> bool` predicate..

Reply:
@fadeevab 
>     1. What would be the best average choice from those crates?

I would probably choose [strsim](https://crates.io/crates/strsim/0.11.1) which has no additional deps and skip "wildcard" tbh. Fuzzy is close-enough to wildcard for a v1 :smile: 

>     2. Let's implement expandability in a separate PR as an extension method like `with_filter()` etc. It will be easier to review and test.

Sounds good :+1:

Reply:
@fadeevab upon further thought, I would probably even consider removing all filtering logic from `cliclack` itself and use pt.2 exclusively...

Reply:
On this part, I will wait for you guys to chose because you kinda lost me 😆

Reply:
@antoinePutmans let it be `strsim`...
@cylewitruk a good enough default is needed for usability.

It doesn't look like a huge dep, it will be optimized out by the compiler. It may slow down a compile time though. It could be under the feature flag, but only if other similar prompt crates put it under the feature flag (let's do it only if it impacts the build time hugely).

Reply:
I changed the filtering to use the strsim::jaro_winkler() function 
The behaviour is a bit different from the previous version so you maybe you guys should try it and tell me if it seems good to you

Reply:
@antoinePutmans Pretty nice :)
![image](https://github.com/fadeevab/cliclack/assets/5967447/93759a12-a8e8-4168-895f-b146f9977a05)

Meantime, I believe an empty line with a cursor should be rendered for the fuzzy mode.